modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -187,7 +188,16 @@ private IgniteClientDisconnectedCheckedException disconnectedError(@Nullable Ign
     /** {@inheritDoc} */
     @SuppressWarnings("TooBroadScope")
     @Override public void onKernalStop(boolean cancel) {
-        lock.writeLock();
+        while(true) {
+            try {
+                if (lock.tryWriteLock(1, TimeUnit.SECONDS))
+                    break;
+                else
+                    Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                U.warn(log, e);
```

Comment:
May be make sense to print a warning "Await for lock on node stop was interrupted"?

Reply:
maybe you're right
thanks